DependencyInjectionContainer/DIContainer.cs

```diff
@@ -3,73 +3,90 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
-using System.ComponentModel.Composition;
 using System.Reflection;
+using DependencyInjectionContainer.Attributes;
+using System.Collections;
+using System.Data;
+using System.Reflection.Metadata;
 
 namespace DependencyInjectionContainer
 {
-    public sealed class DIContainer
+    public sealed record DIContainer
     {
-        private List<Service> _services;
-        private DIContainer _curContainerParent;
+        private DIContainer curContainerParent;
 
-        internal DIContainer(List<Service> services, DIContainer containerParent)
+        internal DIContainer(List<Service> _services, DIContainer containerParent)
         {
-            _services = services;
-            _curContainerParent = containerParent;
+            Services = _services;
+            curContainerParent = containerParent;
         }
+        internal List<Service> Services { get; private set; }
```

Comment:
Why List but not IEnumerable<Service>?
I can clear list or add items even if it private set